Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp

```diff
@@ -349,9 +349,12 @@ void LayerTreeHost::ensureDrawing()
 void LayerTreeHost::sizeDidChange()
 {
     m_pendingResize = true;
-    if (m_isWaitingForRenderer)
+    if (m_isWaitingForRenderer) {
+        // Set m_isWaitingForRenderer to false, otherwise scheduleLayerFlush() will return without
+        // having scheduled anything.
```

Comment:
This is the expected behavior, scheduleLayerFlush returns early but sets m_scheduledWhileWaitingForRenderer so that when the current composition finishes we run another layer flush from LayerTreeHost::didComposite. If that's not happening, we need to understand why.

Reply:
I've spent days sprinkling debug printfs in this code (there is no logging whatsover) on my spare time. 

The conclusion I came up to was that the WebProcess startup is delayed and it doesn't receive the message from the UIProcess regarding the page size... I don't know why.

This patch is fixing the bug I've been experiencing for a couple months now, I don't think I'll get any further, so if anyone wants to take over, please do so :)

Reply:
I really appreciate it, I'm interested in fixing this, but I can't reproduce it, so if you could share your printf logs it would help a lot.

Reply:
https://github.com/philn/WebKit/tree/wip/288602

[good.log](https://github.com/user-attachments/files/19929044/good.log)
[bad.log](https://github.com/user-attachments/files/19929045/bad.log)

Reply:
Thanks, this helped a lot, I think I understand the problem now. This happens when the device scale factor change comes before the web view geometry update. In that case we should not do a layer flush, since the page size is empty. The geometry update will trigger the flush with the right size.

Reply:
Could you try PR #44592 ?

Reply:
OK, I can test it this afternoon when I'm back home.